There is no cleanup stage if mongodump executable not found

Here is the result of sample run:

```
$ mongodb-consistent-backup -n ff -l ~/backup_dir/ -P 21021
[2017-07-27 12:47:29,771] [INFO] [MainProcess] [Main:init:143] Starting mongodb-consistent-backup version 1.0.3 (git commit: 90e7f2fd9cb58bcbdc63bf0d8640e3c7f411e7f0)
[2017-07-27 12:47:29,771] [INFO] [MainProcess] [Main:init:144] Loaded config: {"archive": {"method": "tar", "tar": {"compression": "gzip"}, "zbackup": {"binary": "/usr/bin/zbackup", "cache_mb": 128, "compression": "lzma"}}, "authdb": "admin", "backup": {"location": "/home/shahriyar.rzaev/backup_dir/", "method": "mongodump", "mongodump": {"binary": "/usr/bin/mongodump", "compression": "auto"}, "name": "ff"}, "environment": "production", "host": "localhost", "lock_file": "/tmp/mongodb-consistent-backup.lock", "notify": {"method": "none"}, "oplog": {"compression": "none", "tailer": {"status_interval": 30}}, "port": 21021, "replication": {"max_lag_secs": 5, "max_priority": 1000}, "sharding": {"balancer": {"ping_secs": 3, "wait_secs": 300}}, "upload": {"method": "none", "s3": {"chunk_size_mb": 50, "region": "us-east-1", "retries": 5, "secure": true, "threads": 4}}}
[2017-07-27 12:47:29,771] [INFO] [MainProcess] [Stage:init:32] Notify stage disabled, skipping
[2017-07-27 12:47:29,774] [INFO] [MainProcess] [State:init:135] Initializing root state directory /home/shahriyar.rzaev/backup_dir/ff
[2017-07-27 12:47:29,775] [INFO] [MainProcess] [State:load_backups:153] Found 0 existing completed backups for set
[2017-07-27 12:47:29,775] [INFO] [MainProcess] [State:init:119] Initializing backup state directory: /home/shahriyar.rzaev/backup_dir/ff/20170727_1247
[2017-07-27 12:47:29,776] [INFO] [MainProcess] [Stage:init:32] Upload stage disabled, skipping
[2017-07-27 12:47:29,776] [INFO] [MainProcess] [Main:run:268] Running backup in replset mode using seed node(s): localhost:21021
[2017-07-27 12:47:29,863] [CRITICAL] [MainProcess] [Mongodump:can_gzip:60] Cannot find or execute the mongodump binary file /usr/bin/mongodump!
```

But if it is could not connect to mongo there is a cleanup stage:

```
$ mongodb-consistent-backup -n ff -l /home/shahriyar.rzaev/backup_dir/
[2017-07-27 12:34:44,318] [INFO] [MainProcess] [Main:init:143] Starting mongodb-consistent-backup version 1.0.3 (git commit: 90e7f2fd9cb58bcbdc63bf0d8640e3c7f411e7f0)
[2017-07-27 12:34:44,318] [INFO] [MainProcess] [Main:init:144] Loaded config: {"archive": {"method": "tar", "tar": {"compression": "gzip"}, "zbackup": {"binary": "/usr/bin/zbackup", "cache_mb": 128, "compression": "lzma"}}, "authdb": "admin", "backup": {"location": "/home/shahriyar.rzaev/backup_dir/", "method": "mongodump", "mongodump": {"binary": "/usr/bin/mongodump", "compression": "auto"}, "name": "ff"}, "environment": "production", "host": "localhost", "lock_file": "/tmp/mongodb-consistent-backup.lock", "notify": {"method": "none"}, "oplog": {"compression": "none", "tailer": {"status_interval": 30}}, "port": 27017, "replication": {"max_lag_secs": 5, "max_priority": 1000}, "sharding": {"balancer": {"ping_secs": 3, "wait_secs": 300}}, "upload": {"method": "none", "s3": {"chunk_size_mb": 50, "region": "us-east-1", "retries": 5, "secure": true, "threads": 4}}}
[2017-07-27 12:34:44,318] [INFO] [MainProcess] [Stage:init:32] Notify stage disabled, skipping
[2017-07-27 12:34:49,330] [ERROR] [MainProcess] [DB:connect:47] Unable to connect to localhost:27017! Error: localhost:27017: [Errno 111] Connection refused
[2017-07-27 12:34:49,331] [CRITICAL] [MainProcess] [Main:exception:218] Cannot connect to seed host(s): localhost:27017
[2017-07-27 12:34:49,331] [INFO] [MainProcess] [Main:cleanup_and_exit:174] Starting cleanup procedure! Stopping running threads
[2017-07-27 12:34:49,334] [INFO] [MainProcess] [Main:cleanup_and_exit:204] Cleanup complete, exiting
```

--- a/mongodb_consistent_backup/Backup/Mongodump/Mongodump.py
+++ b/mongodb_consistent_backup/Backup/Mongodump/Mongodump.py
@@ -1,6 +1,5 @@
 import os
 import logging
-import sys
 
 from math import floor
 from subprocess import check_output
@@ -71,8 +70,7 @@ class Mongodump(Task):
                 return True
             return False
         else:
-            logging.fatal("Cannot find or execute the mongodump binary file %s!" % self.binary)
-            sys.exit(1)
+            raise OperationError("Cannot find or execute the mongodump binary file %s!" % self.binary)
 
     def summary(self):
         return self._summary

